firstuseauthenticator/firstuseauthenticator.py

```diff
@@ -108,6 +122,13 @@ def authenticate(self, handler, data):
                 return None
 
         password = data['password']
+        # Don't enforce password length requirement on existing users, since that can
+        # lock users out of their hubs.
+        if not self._validate_password(password) and not self._user_exists(username):
+            self.log.error('Password too short! \
```

Comment:
Does this display the error to the user? If not, is there an easy way for us to do that? That'd be great...

Reply:
The message is displayed to the console. The user only sees "Invalid username or password" and I believe it comes from [here](https://github.com/jupyterhub/jupyterhub/blob/f97ef7eaac73e4af4a5444b24fdffe47f388da70/jupyterhub/handlers/login.py#L153).

From what I understand, in order to have our own error message we could define our own *LoginHandler* and overload the *post* method. What do you think? Do you want me to give this a try?

Reply:
@yuvipanda, I believe a nicer way we could do this is to make the *login_error* a configurable attribute of *LoginHandler* and have it print *Invalid username or password* only if no specific error message is defined. But this implies modifying code in JupyterHub.